float32[] ids
float32[] xpos
float32[] ypos
